mcstatus/status_response.py

```diff
@@ -0,0 +1,667 @@
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build JavaStatusResponse and check is it valid.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (players, version, description) are not present.
+        :raise TypeError: If the required keys (players - dict, version - dict, description - str)
+            are not of the specified type.
+        :return: `JavaStatusResponse` object.
+        """
+        _validate_data(raw, "status", [("players", dict), ("version", dict), ("description", str)])
+        return cls(
+            players=JavaStatusPlayers.build(raw["players"]),
+            version=JavaStatusVersion.build(raw["version"]),
+            motd=cls._parse_description(raw["description"]),
+            icon=raw.get("favicon"),
+            # This will be set later.
+            latency=None,  # type: ignore[assignment]
+        )
+
+    @staticmethod
+    def _parse_description(raw_description: Union[dict, list, str]) -> str:
+        """Parse description from raw response.
+
+        :param raw_description: Raw description.
+        :return: Parsed description.
+        """
+        if isinstance(raw_description, str):
+            return raw_description
+
+        if isinstance(raw_description, dict):
+            entries = raw_description.get("extra", [])
+            end = raw_description["text"]
+        else:
+            entries = raw_description
+            end = ""
+
+        description = ""
+
+        for entry in entries:
+            for style_key, style_val in STYLE_MAP.items():
+                if entry.get(style_key):
+                    try:
+                        if isinstance(style_val, dict):
+                            style_val = style_val[entry[style_key]]
+
+                        description += f"§{style_val}"
+                    except KeyError:
+                        pass  # ignoring these key errors strips out html color codes
+            description += entry.get("text", "")
+
+        return description + end
+
+
+@dataclass
+class NewBedrockStatusResponse(BaseStatusResponse):
+    """Class for storing BedrockServer data from a status response.
+
+    :param map_name: Name of the map. Can be unset.
+    :param gamemode: Gamemode of the server. Can be hidden.
+    """
+
+    players: BedrockStatusPlayers
+    version: BedrockStatusVersion
+    map_name: Optional[str]
+    gamemode: Optional[str]
+
+    @classmethod
+    def build(cls, decoded_data: List[Any], latency: float) -> Self:
+        """Build BaseStatusResponse and check is it valid.
+
+        :param decoded_data: Raw decoded response object.
+        :param latency: Latency of the request.
+        :return: `BedrockStatusResponse` object.
+        """
+
+        try:
+            map_name = decoded_data[7]
+        except IndexError:
+            map_name = None
+        try:
+            gamemode = decoded_data[8]
+        except IndexError:
+            gamemode = None
```

Comment:
While I tried make coverage level to 100% here, I noticed that list which going here is more than 8 items, and there is imposible make `gamemode`/`map_name` to `None`.

Reply:
Why would it be impossible? Just pass a list of 7 elements (max index of 6), which will be failing both try-except blocks. This shouldn't cause any issues since we aren't accessing any index above 5 in the code below this.

You could also pass a list of 8 elements (max index of 7), to test that the `map_name` is set, and only the `gamemode` was `None`.

Reply:
Fixed in 6e9b60d2606bddd1e424ec4f451405f386238235, thanks.